File: src/Core/Api.Common.Fs/Metrics.fs
namespace Api.Common.Fs

open Sentry

module Metrics =

    let startTransactionWithUser (name: string) (operation: string) (userId: int) =
        SentrySdk.ConfigureScope(fun scope -> scope.TransactionName <- name)
        let tr = SentrySdk.StartTransaction(name, operation)
        tr.User <- new SentryUser(Id = userId.ToString())
        tr

File: src/Core/Api.Common/Metrics.cs
namespace Api.Common;

public static class Metrics
{
    public static ITransactionTracer StartTransaction(string name, string operation)
    {
        SentrySdk.ConfigureScope(scope => scope.TransactionName = name);
        return SentrySdk.StartTransaction(name, operation);
    }

    public static ITransactionTracer StartTransactionWithUser(string name, string operation, int userId)
    {
        var tr = StartTransaction(name, operation);
        tr.User = new SentryUser { Id = userId.ToString() };
        tr.Status = SpanStatus.Ok;

        return tr;
    }
}
